src/main/java/uk/gov/companieshouse/itemhandler/service/FilingHistoryDescriptionProviderService.java

```diff
@@ -0,0 +1,121 @@
+package uk.gov.companieshouse.itemhandler.service;
+
+import org.springframework.stereotype.Service;
+import org.yaml.snakeyaml.Yaml;
+import uk.gov.companieshouse.itemhandler.logging.LoggingUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.itemhandler.logging.LoggingUtils.DESCRIPTION_LOG_KEY;
+
+@Service
+public class FilingHistoryDescriptionProviderService {
+
+    private static final String FILING_HISTORY_DESCRIPTIONS_FILEPATH = "api-enumerations/filing_history_descriptions.yml";
+    private static final String FILING_HISTORY_DESCRIPTION_KEY = "description";
+    private static final String LOG_MESSAGE_FILE_KEY = "file";
+
+    private Map<String, String> filingHistoryDescriptions;
+
+    public FilingHistoryDescriptionProviderService() {
+        final File filingHistoryDescriptionsFile = new File(FILING_HISTORY_DESCRIPTIONS_FILEPATH);
+        filingHistoryDescriptions = loadFilingHistoryDescriptionsFromFile(filingHistoryDescriptionsFile);
+    }
+
+    public FilingHistoryDescriptionProviderService(final File filingHistoryDescriptionsFile) {
+        filingHistoryDescriptions = loadFilingHistoryDescriptionsFromFile(filingHistoryDescriptionsFile);
+    }
+
+    private Map<String, String> loadFilingHistoryDescriptionsFromFile(final File filingHistoryDescriptionsFile) {
+        if (!filingHistoryDescriptionsFile.exists()) {
+            Map<String, Object> logMap = new HashMap<>();
+            logMap.put(LOG_MESSAGE_FILE_KEY, filingHistoryDescriptionsFile.getAbsolutePath());
+            LoggingUtils.getLogger().error("Orders descriptions file not found", logMap);
+            return null;
+        }
+        Map<String, String> filingHistoryDescriptionsLocal = null;
+        try(final InputStream inputStream = new FileInputStream(filingHistoryDescriptionsFile)) {
+            final Yaml yaml = new Yaml();
+            final Map<String, Object> filingHistoryDescriptionsRoot = yaml.load(inputStream);
+            filingHistoryDescriptionsLocal =
+                    (Map<String, String>) filingHistoryDescriptionsRoot.get(FILING_HISTORY_DESCRIPTION_KEY);
+            if (filingHistoryDescriptionsLocal == null) {
+                Map<String, Object> logMap = new HashMap<>();
+                logMap.put(DESCRIPTION_LOG_KEY, FILING_HISTORY_DESCRIPTION_KEY);
+                LoggingUtils.getLogger().error("Filing History descriptions file not found", logMap);
+                return null;
+            }
+        } catch (IOException ioe) {
```

Comment:
Should there at least be a log statement for help debugging in the worst case scenario?

Reply:
I copied this over from certified copies. But yeh there probably should be. I'll add it in now.